### ISSUE TITLE
Unclosed attributes disappear instead of turning into regular text

### ISSUE THREAD

**Issue by hellux** [jgm/djot#126]
E.g.

```markdown
text{a=x

{a=x
# heading
```

generates

```html
<p>text</p>
<p></p>
<section id="heading">
<h1>heading</h1>
</section>
```

instead of

```html
<p>text{a=x</p>
<p>{a=x
# heading</p>
```

Similar to #65

**Comment by jgm** [comment]
With this push we get

```
<p>text{a=x</p>
<p>{a=x</p>
<section id="heading">
<h1>heading</h1>
</section>
```

Still not quite right.  If `{a=x` before `# heading` is just regular text, then we shouldn't get a heading at all.

We need to put some code into the block parser that is similar to what I just added to the inline parser.
Better, make this backtracking code a method on the attribute parser.

**Comment by jgm** [comment]
Reverting for now because this commit caused fuzz test failures.
Need to rethink.

**Comment by jgm** [comment]
Closed by c68e807cc3343364b2570c6f69a8eaeeb17d0ece
and earlier commits.
